testing/src/KokkosFFT_AlmostEqual.hpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: (C) The kokkos-fft development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0 WITH LLVM-exception
+
+#ifndef KOKKOSFFT_ALMOST_EQUAL_HPP
+#define KOKKOSFFT_ALMOST_EQUAL_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace KokkosFFT {
+namespace Testing {
+namespace Impl {
+
+template <typename ScalarA, typename ScalarB, typename ScalarTol>
+KOKKOS_INLINE_FUNCTION bool are_almost_equal(ScalarA a, ScalarB b,
+                                             ScalarTol rtol, ScalarTol atol) {
+  return Kokkos::abs(a - b) <= (atol + rtol * Kokkos::abs(b));
```

Comment:
Usually this function performs:
```cpp
auto abs_diff = Kokkos::abs(a-b);
if (abs_diff <  atol)
  return true;
abs_ref = f(Kokkos::abs(a), Kokkos::abs(b)); //f = min, max, only a, only b
return abs_diff < rtol*abs_ref;
```

Adding the two tolerances together might lead to weird results, especially if the tolerances are negative.

Reply:
Sure. 
I will fix this